bin/commands/init/stc/index.sh

```diff
@@ -101,157 +82,56 @@ for mb in ${target_proclibs}; do
-  tmpfile=$(create_tmp_file $(echo "zwe ${ZWE_CLI_COMMANDS_LIST}" | sed "s# #-#g"))
-  print_debug "- Copy ${prefix}.${ZWE_PRIVATE_DS_SZWESAMP}(ZWESLSTC) to ${tmpfile}"
-  if [[ "$ZWE_CLI_PARAMETER_CONFIG" != /* ]];then
-    print_message "CONFIG path defined in ZWESLSTC is converted into absolute path and may contain SYSNAME."
-    print_message "Please manually verify if this path works for your environment, especially when you are working in Sysplex environment."
-  fi
-  result=$(cat "//'${prefix}.${ZWE_PRIVATE_DS_SZWESAMP}(ZWESLSTC)'" | \
-          sed "s/^\/\/STEPLIB .*\$/\/\/STEPLIB  DD   DSNAME=${authLoadlib},DISP=SHR/" | \
-          sed "s#^CONFIG=.*\$#CONFIG=$(convert_to_absolute_path ${ZWE_CLI_PARAMETER_CONFIG})#" \
-          > "${tmpfile}")
-  code=$?
-  chmod 700 "${tmpfile}"
-  if [ ${code} -eq 0 ]; then
-    print_debug "  * Succeeded"
-    print_trace "  * Exit code: ${code}"
-    print_trace "  * Output:"
-    if [ -n "${result}" ]; then
-      print_trace "$(padding_left "${result}" "    ")"
-    fi
-  else
-    print_debug "  * Failed"
-    print_error "  * Exit code: ${code}"
-    print_error "  * Output:"
-    if [ -n "${result}" ]; then
-      print_error "$(padding_left "${result}" "    ")"
-    fi
-  fi
-  if [ ! -f "${tmpfile}" ]; then
-    print_error_and_exit "Error ZWEL0159E: Failed to modify ${prefix}.${ZWE_PRIVATE_DS_SZWESAMP}(ZWESLSTC)" "" 159
-  fi
-  print_trace "- ensure ${tmpfile} encoding before copying into data set"
-  ensure_file_encoding "${tmpfile}" "SPDX-License-Identifier"
-  print_trace "- ${tmpfile} created, copy to ${jcllib}(${security_stcs_zowe})"
-  copy_to_data_set "${tmpfile}" "${jcllib}(${security_stcs_zowe})" "" "${ZWE_CLI_PARAMETER_ALLOW_OVERWRITE}"
-  code=$?
-  print_trace "- Delete ${tmpfile}"
-  rm -f "${tmpfile}"
-  if [ ${code} -ne 0 ]; then
-    print_error_and_exit "Error ZWEL0160E: Failed to write to ${jcllib}(${security_stcs_zowe}). Please check if target data set is opened by others." "" 160
-  fi
-  print_debug "- ${jcllib}(${security_stcs_zowe}) is prepared"
 
-  # ZWESISTC
-  print_message "Modify ZWESISTC and save as ${jcllib}(${security_stcs_zis})"
-  tmpfile=$(create_tmp_file $(echo "zwe ${ZWE_CLI_COMMANDS_LIST}" | sed "s# #-#g"))
-  print_debug "- Copy ${prefix}.${ZWE_PRIVATE_DS_SZWESAMP}(ZWESISTC) to ${tmpfile}"
-  result=$(cat "//'${prefix}.${ZWE_PRIVATE_DS_SZWESAMP}(ZWESISTC)'" | \
-          sed '/^..STEPLIB/c\
-\//STEPLIB  DD   DSNAME='${authLoadlib}',DISP=SHR\
-\//         DD   DSNAME='${authPluginLib}',DISP=SHR' | \
-          sed "s/^\/\/PARMLIB .*\$/\/\/PARMLIB  DD   DSNAME=${parmlib},DISP=SHR/" \
-          > "${tmpfile}")
-  code=$?
-  chmod 700 "${tmpfile}"
-  if [ ${code} -eq 0 ]; then
-    print_debug "  * Succeeded"
-    print_trace "  * Exit code: ${code}"
-    print_trace "  * Output:"
-    if [ -n "${result}" ]; then
-      print_trace "$(padding_left "${result}" "    ")"
+  jcl_file=$(create_tmp_file)
+  copy_mvs_to_uss "${jcllib}(ZWEISTC)" "${jcl_file}"
```

Comment:
This will not work for `$` in the dataset name: e.g. `jcllib='ZOWE.$JCLLIB'`, gets expanded and most likely tries to copy from `ZOWE.(ZWEISTC)`.

Reply:
i guess now that the code is starting to work, just migrating it over to the TS code will help